tests/components/influxdb/test_init.py

```diff
@@ -11,748 +11,1116 @@
+
+
+def _get_write_api_mock_v1(mock_influx_client):
+    """Return the write api mock for the V1 client."""
+    return mock_influx_client.return_value.write_points
+
+
+def _get_write_api_mock_v2(mock_influx_client):
+    """Return the write api mock for the V2 client."""
+    return mock_influx_client.return_value.write_api.return_value.write
+
+
+@pytest.mark.parametrize(
+    "mock_client, config_ext, get_write_api",
+    [
+        (
+            influxdb.DEFAULT_API_VERSION,
+            {
+                "api_version": influxdb.DEFAULT_API_VERSION,
                 "username": "user",
                 "password": "password",
-                "max_retries": 4,
-                "ssl": "False",
                 "verify_ssl": "False",
-            }
+            },
+            _get_write_api_mock_v1,
+        ),
+        (
+            influxdb.API_VERSION_2,
+            {
+                "api_version": influxdb.API_VERSION_2,
+                "token": "token",
+                "organization": "organization",
+                "bucket": "bucket",
+            },
+            _get_write_api_mock_v2,
+        ),
+    ],
+    indirect=["mock_client"],
+)
+# pylint: disable=redefined-outer-name
+async def test_setup_config_full(hass, mock_client, config_ext, get_write_api):
+    """Test the setup with full configuration."""
+    config = {
+        "influxdb": {
+            "host": "host",
+            "port": 123,
+            "database": "db",
+            "max_retries": 4,
+            "ssl": "False",
         }
-        assert setup_component(self.hass, influxdb.DOMAIN, config)
-        assert self.hass.bus.listen.called
-        assert EVENT_STATE_CHANGED == self.hass.bus.listen.call_args_list[0][0][0]
-        assert mock_client.return_value.write_points.call_count == 1
-
-    def test_setup_config_defaults(self, mock_client):
-        """Test the setup with default configuration."""
-        config = {"influxdb": {"host": "host", "username": "user", "password": "pass"}}
-        assert setup_component(self.hass, influxdb.DOMAIN, config)
-        assert self.hass.bus.listen.called
-        assert EVENT_STATE_CHANGED == self.hass.bus.listen.call_args_list[0][0][0]
-
-    def test_setup_minimal_config(self, mock_client):
-        """Test the setup with minimal configuration."""
-        config = {"influxdb": {}}
-
-        assert setup_component(self.hass, influxdb.DOMAIN, config)
-
-    def test_setup_missing_password(self, mock_client):
-        """Test the setup with existing username and missing password."""
-        config = {"influxdb": {"username": "user"}}
-
-        assert not setup_component(self.hass, influxdb.DOMAIN, config)
-
-    def _setup(self, mock_client, **kwargs):
-        """Set up the client."""
-        config = {
-            "influxdb": {
-                "host": "host",
+    }
+    config["influxdb"].update(config_ext)
+
+    assert await async_setup_component(hass, influxdb.DOMAIN, config)
```

Comment:
We need to block till done to complete the set up tasks.
```suggestion
    assert await async_setup_component(hass, influxdb.DOMAIN, config)
    await hass.async_block_till_done()
```

Reply:
Ok so I think I have to fix this one in a few places, will do that.

Also I notice you're using `async_block_till_done`. Inside the rest of the tests I had been doing this since its what the original author was doing:
```python
hass.data[influxdb.DOMAIN].block_till_done()
```
Seems like I should also switch those to this, right?
```python
await hass.data[influxdb.DOMAIN].async_block_till_done()
```

Reply:
Ok scratch that. I see now, that gets an `InfluxThread` object which does not have an `async_block_till_done` method. Just changing the first two tests that were missing the logic to block until done.

Reply:
The thread block till done joins the queue of the thread. It doesn't do the same thing as `hass.async_block_till_done` so I don't think we can assume we can avoid calling the latter.

Reply:
Ok so are you saying any place place with this line:
```
hass.data[influxdb.DOMAIN].block_till_done()
```
should be changed to this?
```diff
hass.data[influxdb.DOMAIN].block_till_done()
+ await hass.async_block_till_done()
```

Reply:
No, these two block till done methods have different purposes.

The core method is used to wait until all tasks that have been scheduled via the core are done. We need to use that after setting up the integration with `async_setup_component`.

The influxdb thread `block_till_done` is used to wait until all the events have been processed from the queue.

I haven't checked yet if we need to add `hass.async_block_till_done` in more places.

Reply:
Yea that makes sense. Ok let me know. And thanks for all your help!